packages/map/src/marker/index.jsx

```diff
@@ -13,37 +13,47 @@ const Marker = ({
   popupDefaultState,
   popupContent,
   popupOffset,
-  ...rest
 }) => {
   const nodeRef = useRef();
   const popupRef = useRef();
-  const map = useContext(MapContext);
-  useChildrenBounds([position]);
+  const [, forceRender] = useState();
 
-  useEffect(() => {
-    if (!map) return;
+  useChildrenBounds([position]);
 
+  useMapEffect((map) => {
+    nodeRef.current = document.createElement('div');
```

Comment:
If we really need to create the element multiple times, then add a comment (see line 29)